### PR TITLE
Fix memory leak in ConstConstraint

### PR DESCRIPTION
**Pull request by oppm**
This issue should up when we run some tests with enabled address sanitizer.

### CODE REVIEW

File: include/valijson/constraints/concrete_constraints.hpp [hunk 1]
```diff
@@ -192,6 +192,11 @@ class ConstConstraint: public BasicConstraint<ConstConstraint>
       : BasicConstraint(other),
         m_value(other.m_value->clone()) { }
 
+    ~ConstConstraint() override
+    {
+        delete m_value;
```
Comment (tristanpenman) [inline include/valijson/constraints/concrete_constraints.hpp:197]: I don't entirely follow why this is required. `m_value` is a `std::unique_ptr`, so I would have expected the `FrozenValue` destructor to be called, and its memory to be freed, when the ConstConstraint class is destroyed.

Comment (tristanpenman) [reply include/valijson/constraints/concrete_constraints.hpp:197]: This actually doesn't compile. Is there something I'm missing here?

Comment (oppm) [reply include/valijson/constraints/concrete_constraints.hpp:197]: Ah, I see, we are using an older version. In our version this change fixed the memory leak. I will close this pull request. Sorry for the hassle.

**Comment by oppm** [comment]
Wrong change.

**Pull request closed without merging**